src/output.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Output helpers.
+
+/// Gets a colorized message for outputting to a particular stream. Do not use colors if the stream
+/// is not a tty.
+#[macro_export]
+macro_rules! colorize {
+    ($color:ident, $stream:ident, $arg:tt) => {{
+        let style = if atty::is(atty::Stream::$stream) {
+            ansi_term::Color::$color.normal()
+        } else {
+            ansi_term::Style::default()
+        };
+        style.paint($arg)
+    }};
+    ($color:ident, $stream:ident, $($arg:tt)*) => {{
+        let style = if atty::is(atty::Stream::$stream) {
+            ansi_term::Color::$color.normal()
+        } else {
+            ansi_term::Style::default()
+        };
+        style.paint(format!($($arg)*))
+    }};
+}
+
+/// Wrapper around `print!` that accepts a color as its first argument.
+#[macro_export]
+macro_rules! print_colored {
+    ($color:ident, $($arg:tt)*) => {
+        print!("{}", colorize!($color, Stdout, $($arg)*))
+    };
+}
+
+/// Wrapper around `println!` that accepts a color as its first argument.
+#[macro_export]
+macro_rules! println_colored {
+    ($color:ident, $($arg:tt)*) => {
+        println!("{}", colorize!($color, Stdout, $($arg)*))
+    };
+}
+
+/// Wrapper around `eprint!` that accepts a color as its first argument.
+#[macro_export]
+macro_rules! eprint_colored {
+    ($color:ident, $($arg:tt)*) => {
+        eprint!("{}", colorize!($color, Stderr, $($arg)*))
+    };
+}
+
+/// Wrapper around `eprintln!` that accepts a color as its first argument.
+#[macro_export]
+macro_rules! eprintln_colored {
+    ($color:ident, $($arg:tt)*) => {
+        eprintln!("{}", colorize!($color, Stderr, $($arg)*))
+    };
+}
+
+/// Write an info message to stdout.
+#[macro_export]
+macro_rules! info {
+    ($($arg:tt)*) => {
+        print!("{}", colorize!(Blue, Stdout, "[INFO] "));
+        println!($($arg)*);
+    };
+}
+
+/// Write an error message to stdout.
+#[macro_export]
+macro_rules! err {
+    ($($arg:tt)*) => {
+        print!("{}", colorize!(Red, Stdout, "[ERR] "));
+        println!($($arg)*);
+    };
+}
+
+/// Write a success message to stdout.
+#[macro_export]
+macro_rules! success {
+    ($($arg:tt)*) => {
+        print!("{}", colorize!(Green, Stdout, "[SUCCESS] "));
+        println!($($arg)*);
+    };
+}
+
+/// Write an info message to stderr.
+#[macro_export]
+macro_rules! e_info {
+    ($($arg:tt)*) => {
+        eprint!("{}", colorize!(Blue, Stderr, "[INFO] "));
+        eprintln!($($arg)*);
+    };
+}
+
+/// Write an error message to stderr.
+#[macro_export]
+macro_rules! e_err {
+    ($($arg:tt)*) => {
+        eprint!("{}", colorize!(Red, Stderr, "[ERR] "));
+        eprintln!($($arg)*);
+    };
+}
+
+/// Write a success message to stderr.
+#[macro_export]
+macro_rules! e_success {
+    ($($arg:tt)*) => {
+        eprint!("{}", colorize!(Green, Stderr, "[SUCCESS] "));
+        eprintln!($($arg)*);
+    };
+}
```

Comment:
I'm somewhat confused by this - why write a success message to stderr?

Reply:
AFAIK stderr is not just used for error messages, but diagnostic messages that aren't part of a program's 'functional output' too. It's definitely possible to have a success message that's just a diagnostic message, which is why the option is available. 😄

Reply:
That's definitely good to know, thanks! A very unfortunate naming choice for the stream, though 😄